barman/recovery_executor.py

```diff
@@ -1807,13 +1810,318 @@ def _backup_copy(self, backup_info, dest, remote_command=None, **kwargs):
+        :return Iterator[barman.infofile.LocalBackupInfo]: iterator of paths of
+            the backups in the chain, going from the full to the incremental backup
+            pointed by *backup_info*
+        """
+        return reversed(
+            [backup.get_data_directory() for backup in backup_info.walk_to_root()]
+        )
+
+    def _prepare_destination(self, dest_dir):
+        """
+        Prepare the destination directory or file before moving it.
+
+        This method is responsible for removing a directory if it already
+        exists, then (re)creating it and ensuring the correct permissions
+        on the directory.
+
+        :param str dest_dir: destination directory
+        """
+        # Remove a dir if exists. Ignore eventual errors
+        shutil.rmtree(dest_dir, ignore_errors=True)
+        # create the dir
+        mkpath(dest_dir)
+        # Ensure the right permissions to the destination directory
+        # chmod 0700 octal
+        os.chmod(dest_dir, 448)
+
+    def _retry_handler(self, dest_dirs, attempt):
+        """
+        Handler invoked during a combine backup in case of retry.
+
+        The method simply warn the user of the failure and
+        remove the already existing directories of the backup.
+
+        :param list[str] dest_dirs: destination directories
+        :param int attempt: attempt number (starting from 0)
+        """
+        output.warning(
+            "Failure combining backups using pg_combinebackup (attempt %s)", attempt
+        )
+        output.warning(
+            "The files created so far will be removed and "
+            "the combine process will restart in %s seconds",
+            "30",
+        )
+        # Remove all the destination directories and reinit the backup
+        for _dir in dest_dirs:
+            self._prepare_destination(_dir)
+
+    def _fetch_remote_status(self):
+        """
+        Gather info from the remote server.
+
+        This method does not raise any exception in case of errors,
+        but set the missing values to ``None`` in the resulting dictionary.
+
+        :return dict[str, str|bool]: the pg_combinebackup client information
+            of the remote server.
+        """
+        remote_status = dict.fromkeys(
+            (
+                "pg_combinebackup_installed",
+                "pg_combinebackup_path",
+                "pg_combinebackup_version",
+            ),
+            None,
+        )
+
+        # Test pg_combinebackup existence
+        version_info = PgCombineBackup.get_version_info(self.server.path)
+
+        if version_info["full_path"]:
+            remote_status["pg_combinebackup_installed"] = True
+            remote_status["pg_combinebackup_path"] = version_info["full_path"]
+            remote_status["pg_combinebackup_version"] = version_info["full_version"]
+        else:
+            remote_status["pg_combinebackup_installed"] = False
+            return remote_status
```

Comment:
```suggestion
            return remote_status
```

Doesn't add value. The next instructions is exactly the same.